zcash_keys/src/keys.rs

```diff
@@ -928,6 +928,351 @@ impl UnifiedFullViewingKey {
     }
 }
 
+/// A [ZIP 316](https://zips.z.cash/zip-0316) unified incoming viewing key.
+#[derive(Clone, Debug)]
+pub struct UnifiedIncomingViewingKey {
+    #[cfg(feature = "transparent-inputs")]
+    transparent: Option<zcash_primitives::legacy::keys::ExternalIvk>,
+    #[cfg(feature = "sapling")]
+    sapling: Option<::sapling::zip32::IncomingViewingKey>,
+    #[cfg(feature = "orchard")]
+    orchard: Option<orchard::keys::IncomingViewingKey>,
+    /// Stores the unrecognized elements of the unified encoding.
+    unknown: Vec<(u32, Vec<u8>)>,
+}
+
+impl TryFrom<UnifiedFullViewingKey> for UnifiedIncomingViewingKey {
```

Comment:
Instead of `TryFrom`, this should just be a method on `UnifiedFullViewingKey`.

Reply:
Heh heh. Ya, I wrote this before reading your feelings on the From trait on discord last night. I figured you'd comment on this if I didn't change it first. That's fine.